### PR TITLE
number of prefetched cache hit added

### PR DESCRIPTION
**Pull request by yunyun3599**

### CODE REVIEW

File: cache.py [hunk 1]
```diff
@@ -4,10 +4,12 @@
 class Cache:
     def __init__(self, disk):
         self.pool = {}
+        self.prefetched={}
```
Comment (cezanne) [inline cache.py]: apply coding style
`self.prefetched = {}`

File: storage_default.py [hunk 2]
```diff
@@ -39,5 +39,6 @@ def prefetch(self, pid, lba_base, nblks):
     def report(self):
         print("Req Read:{}, Req Write:{}".format(self.n_accs_read, self.n_accs_write))
         print("# of prefetch: {}".format(self.n_prefetch))
+        print("# of prefetch hit : {}".format(self.cache.n_prefetch_hit))
```
Comment (cezanne) [inline storage_default.py]: Please present `prefetch accuracy`(%) with total and hit counts.

File: cache.py [hunk 3]
```diff
@@ -36,6 +40,7 @@ def prefetch(self, pid, lba_base, nblks):
             if not lba in self.pool:
                 self.disk.read(lba)
                 self.replace_cache(pid, lba)
+                self.prefetched[lba]=True
```
Comment (cezanne) [inline cache.py]: when does the inserted lba go out from the prefetched dictionary  ?

**Review by cezanne — COMMENTED** [review]
Please improve and amend commit..

File: cache.py [hunk 4]
```diff
@@ -4,10 +4,13 @@
 class Cache:
     def __init__(self, disk):
         self.pool = {}
+        self.prefetched = {}
```
Comment (yunyun3599) [inline cache.py:7]: dictionary which stores prefetched address

File: cache.py [hunk 5]
```diff
@@ -24,8 +27,13 @@ def load(self, pid, lba_base, nblks):
             if lba in self.pool:
                 self.policy.hit(lba)
                 self.n_hits += 1
+                if lba in self.prefetched:
```
Comment (yunyun3599) [inline cache.py:29]: when cache hit occurs, if the address is cached by prefetching then increase n_prefetch hit.

File: cache.py [hunk 6]
```diff
@@ -4,10 +4,12 @@
 class Cache:
     def __init__(self, disk):
         self.pool = {}
+        self.prefetched = {}
         self.n_blks_cached_max = 0
         self.n_blks_cached = 0
         self.n_hits = 0
         self.n_miss = 0
+        self.n_prefetch_hit = 0
```
Comment (cezanne) [inline cache.py]: `n_hits_prefetch` name looks better. It's a harmony with `n_hits`.

File: cache.py [hunk 7]
```diff
@@ -24,6 +26,10 @@ def load(self, pid, lba_base, nblks):
             if lba in self.pool:
                 self.policy.hit(lba)
                 self.n_hits += 1
+                if lba in self.prefetched:
+                    if self.prefetched[lba] == False:
```
Comment (cezanne) [inline cache.py]: how about ? `if not self.prefetched[lba] :`

**Pull request merged**